File: frontend/src/types/steam.ts
export interface SteamProfile {
  steam_id: string
  username: string
  avatar: string
  profile_url: string
  time_created: number | null
}

export interface GameEntry {
  app_id: number
  name: string
  total_playtime_minutes: number
  total_playtime_hours: number
  owned_by_accounts: { account_index: number; playtime_minutes: number }[]
  img_icon_url: string
}

export interface Progression {
  level: number
  title: string
  current_hours: number
  hours_for_current_level: number
  hours_for_next_level: number
  progress_percent: number
}

export interface GameLegacyAchievement {
  id: string
  name: string
  description: string
  achieved: boolean
  progress_current: number
  progress_target: number
  progress_percent: number
}

export interface CombinedIdentity {
  accounts: SteamProfile[]
  total_games: number
  unique_games: number
  total_playtime_hours: number
  library: GameEntry[]
  progression: Progression
  gamelegacy_achievements: GameLegacyAchievement[]
}

export interface AchievementEntry {
  api_name: string
  name: string
  description: string
  achieved: boolean
  unlock_time: number
  icon: string
  icon_gray: string
}

export interface GameAchievements {
  has_achievements: boolean
  achieved: number
  total: number
  achievements: AchievementEntry[]
  message: string
}

File: backend/app/schemas/steam.py
from pydantic import BaseModel


class SteamAccountInput(BaseModel):
    steam_id: str


class MultiAccountInput(BaseModel):
    steam_ids: list[str]


class GameEntry(BaseModel):
    app_id: int
    name: str
    total_playtime_minutes: int
    total_playtime_hours: float
    owned_by_accounts: list[dict]
    img_icon_url: str


class PlayerSummary(BaseModel):
    steam_id: str
    username: str
    avatar: str
    profile_url: str
    time_created: int | None = None


class ProgressionOut(BaseModel):
    level: int
    title: str
    current_hours: float
    hours_for_current_level: float
    hours_for_next_level: float
    progress_percent: float


class GameLegacyAchievementOut(BaseModel):
    id: str
    name: str
    description: str
    achieved: bool
    progress_current: float
    progress_target: float
    progress_percent: float


class CombinedIdentity(BaseModel):
    accounts: list[PlayerSummary]
    total_games: int
    unique_games: int
    total_playtime_hours: float
    library: list[GameEntry]
    progression: ProgressionOut
    gamelegacy_achievements: list[GameLegacyAchievementOut]


class AchievementEntry(BaseModel):
    api_name: str
    name: str
    description: str
    achieved: bool
    unlock_time: int
    icon: str
    icon_gray: str


class GameAchievements(BaseModel):
    has_achievements: bool
    achieved: int
    total: int
    achievements: list[AchievementEntry]
    message: str = ""